### PR TITLE
Upgrade Mongoid and associated gems

### PR DESCRIPTION
**Pull request by danielroseman**
This is dependent on a branch of govuk_content_models which upgrades Mongoid
to the latest version. When all apps have been migrated to this version we
can do a release and go back to the versioned gem.

This will need changes to the puppet and deployment scripts to set the new
MONGODB_URI and SECRET_KEY_BASE environment variables.

### CODE REVIEW

File: test/requests/licence_application_request_test.rb [hunk 1]
```diff
@@ -116,9 +107,10 @@ class LicenceApplicationRequestTest < GovUkContentApiTest
   it "should not query the licence api if no licence identifier is present" do
     stub_artefact = FactoryGirl.create(:artefact, slug: 'licence-artefact', state: 'live')
     stub_licence = FactoryGirl.build(:licence_edition, panopticon_id: stub_artefact.id, licence_identifier: nil, state: 'published')
+    stub_licence.save(validate: false)
 
-    Artefact.stubs(:where).with(slug: 'licence-artefact').returns([stub_artefact])
-    Edition.stubs(:where).with(panopticon_id: stub_artefact.id, state: 'published').returns([stub_licence])
+    #Artefact.stubs(:where).with(slug: 'licence-artefact').returns([stub_artefact])
```
Comment (benilovj) [inline test/requests/licence_application_request_test.rb]: Is there a good reason to keep these comments?

Comment (danielroseman) [inline test/requests/licence_application_request_test.rb]: No, I just missed deleting them. Removed and rebased.

**Comment by benilovj** [comment]
@danielroseman I've added linting with 	3b97a39 and 2c4738f. It's made this PR a fair bit noisier, so it may be worth moving those commits to a subsequent PR – your call...

**Comment by boffbowsh** [comment]
This LGTM. @danielroseman are the deployment changes made already?

**Comment by danielroseman** [comment]
@boffbowsh
https://github.gds/gds/alphagov-deployment/pull/1166
https://github.com/alphagov/govuk-puppet/pull/4269 
https://github.gds/gds/deployment/pull/945

**Comment by boffbowsh** [comment]
puppet / deployment changes merged, needs a puppet deploy first

**Comment by benilovj** [comment]
:tada:

**Pull request merged**